internal/proxy/meta_cache.go

```diff
@@ -1016,6 +1007,15 @@ func (m *MetaCache) GetShards(ctx context.Context, withCache bool, database, col
 	// and create new client for new leaders
 	_ = m.shardMgr.UpdateShardLeaders(oldLeaders, ret)
```

Comment:
please remove the UpdateShardLeaders function and init the shartClient in getClient

Reply:
refine the shard cache in next pr